packages/docusaurus-theme-redoc/src/utils/useSpec.ts

```diff
@@ -17,11 +17,11 @@ let currentStore: AppStore | null = null;
  * Released under the MIT License
  */
 export function useSpec(
-  { spec, url, themeId }: SpecProps,
+  { spec, url, themeId, normalizeUrl }: SpecProps,
   optionsOverrides?: RedocRawOptions,
 ) {
   const specOptions = useSpecOptions(themeId, optionsOverrides);
-  const fullUrl = useBaseUrl(url, { absolute: true });
+  const fullUrl = normalizeUrl ? useBaseUrl(url, { absolute: true }) : url;
```

Comment:
No conditional hooks to follow rules of react
```suggestion
  const absoluteUrl = useBaseUrl(url, { absolute: true });
  const fullUrl = specProps.normalizeUrl ? absoluteUrl : url;
```